src/regex_normalization.cpp

```diff
@@ -18,8 +18,20 @@ namespace {
  * @return std::string Reformatted replace pattern
  */
 std::string reformat_replace_pattern(std::string replace_pattern) {
-    return std::regex_replace(replace_pattern, std::regex(R"((\\)([0-9]+))"), R"($$2)");
+    return std::regex_replace(replace_pattern, std::regex(R"((?:\\)([0-9]+))"), R"($$1)");
+}
 
+/**
+ * @brief Fix old search pattern for backward compatibility
+ *
+ * @param search_pattern Search pattern to replace
+ * @return std::string Replaced search pattern
+ */
+std::string fix_search_pattern(std::string search_pattern) {
+    if (search_pattern == R"( ([\\.\\?\\!,])| ('[ms])| (') | ('[rv]e)| (n't))") {
```

Comment:
if there is pattern `r" ([\.\?\!\,])| ('[ms])| (') | ('[rv]e)"` this replace would not work, is it expected?

Reply:
This regex is for detokenization only and is not used anywhere. So I just deleted it.